module/zcommon/zfs_fletcher.c

```diff
@@ -778,6 +779,92 @@ fletcher_4_fini(void)
 	}
 }
 
+/* ABD adapters */
+
+static void
+abd_fletcher_4_init(zio_abd_checksum_data_t *cdp)
+{
+	const fletcher_4_ops_t *ops = fletcher_4_impl_get();
+	cdp->acd_private = (void *) ops;
+
+	ASSERT(ops);
```

Comment:
Let's drop this ASSERT, we've just called `fletcher_4_impl_get()` which has an identical check.